linode/lke/framework_datasource.go

```diff
@@ -97,7 +98,25 @@ func (r *DataSource) Read(
 		return
 	}
 
-	resp.Diagnostics.Append(data.parseLKEAttributes(ctx, cluster, pools, kubeconfig, endpoints, dashboard)...)
+	acl, err := client.GetLKEClusterControlPlaneACL(ctx, clusterId)
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok &&
+			lerr.Code == 400 && strings.Contains(lerr.Message, "Cluster does not support Control Plane ACL") {
```

Comment:
Because ACL isn't returned from the GET cluster endpoint, we have to make an extra call here to get the control plane ACL. There is a tricky case that if this cluster doesn't have an ACL, which is optional, calling GET ACL endpoint will return a 400 and a corresponding error message. I don't think this error should break the data source or resource since ACL is totally optional, so I build it into a warning and continue. Any comment is welcomed!

Reply:
Same as [my other comment](https://github.com/linode/terraform-provider-linode/pull/1436/files#r1608461131), since the ACL not being available on a cluster is a valid case I think it's fine to drop the warning here ^